### PR TITLE
[Doc] Refactor global settings

### PR DESCRIPTION
**Pull request by neozhaoliang**

### CODE REVIEW

**Review by yuanming-hu — COMMENTED** [review]
Thanks, but there are still many undocumented global settings: https://github.com/taichi-dev/taichi/blob/9d27b94ec04943528c4028da4f50fa6dfc25d000/taichi/python/export_lang.cpp#L138-L241

You may want to pick a subset of frequently used options. E.g., `cpu_max_num_threads` is one option that is frequently used but not documented.

Related logic: https://github.com/taichi-dev/taichi/blob/760d501be883dd0ea1c424be186fbbd00aae0dcd/python/taichi/lang/misc.py#L313

**Comment by Vissidarte-Herman** [comment]
I'm still not quite clear as to what "global" means here. Can either of you shed more light on this?

**Comment by neozhaoliang** [comment]
@Vissidarte-Herman Added some explanations for the term "global".

File: docs/lang/articles/reference/global_settings.md [hunk 1]
```diff
@@ -4,72 +4,197 @@ sidebar_position: 3
 
 # Global Settings
 
+
+The call `ti.init()` is indispensable in every Taichi program. It allows you to customize your Taichi runtime by passing arguments to it or setting the corresponding environment variables. Each argument or environment variable will control one specific behavior of the Taichi runtime. For example, the argument `arch` for specifying the backend, the argument `debug` for running the program in debug mode. We have introduced this function in our very first article ["getting started"](../get-started/index.md), but what given there was only a coarse sketch. In this article we will show its full functionalities.
+
+In summary, when calling `ti.init()`, Taichi will execute the following steps to initialize a specific configuration: (we use the `arch` argument to illustrate this)
```
Comment (Olinaaaloompa) [inline docs/lang/articles/reference/global_settings.md:10]:
```suggestion
Generally, when calling `ti.init()`, Taichi executes the following steps to initialize a specific configuration. We use the `arch` argument as an example:
```

**Review by Olinaaaloompa — CHANGES_REQUESTED** [review]

**Review by Olinaaaloompa — CHANGES_REQUESTED** [review]

File: docs/lang/articles/reference/global_settings.md [hunk 2]
```diff
@@ -4,72 +4,197 @@ sidebar_position: 3
...
+        Enable/disable using variables as index to access vector/matrix elements in the Taichi scope.
+
+    kernel_profiler: bool
+        Turn on/off kernel performance profiling.
+
+    offline_cache: bool
+        Enable/disable offline cache of the compiled kernels.
+
+    offline_cache_file_path: str
+        Directory holding the offline cached files.
```
[the remark this answers is not in the document]
Comment (neozhaoliang) [reply docs/lang/articles/reference/global_settings.md:73]: @Olinaaaloompa Should change set a directory -> specify a directory

File: docs/lang/articles/reference/global_settings.md [hunk 3]
```diff
@@ -4,72 +4,197 @@ sidebar_position: 3
...
+[Develop Options]
+
+    gdb_trigger: bool
+        Enable/disable triggering GDB when Taichi crashes, e.g. `ti.init(gdb_trigger=True)`.
+```
+
+
+**Customize via environment variables**
+
+Below are some environment variables that you can set to customize your Taichi program, they overlap but are not in one-to-one correspondence with the `ti.init()` arguments listed above:
```
[the remark this answers is not in the document]
Comment (neozhaoliang) [reply docs/lang/articles/reference/global_settings.md:100]: @Olinaaaloompa  The meaning of the original version is changed: some environment settings do not map to `init` arguments.

File: docs/lang/articles/reference/global_settings.md [hunk 4]
```diff
@@ -4,72 +4,197 @@ sidebar_position: 3
...
+        Set the logging level.
+
+    verbose: bool
+        Eliminate verbose outputs. e.g. `ti.init(verbose=False)`.
+
+
+[Develop Options]
+
+    gdb_trigger: bool
+        Enable/disable triggering GDB when Taichi crashes, e.g. `ti.init(gdb_trigger=True)`.
```
[the remark this answers is not in the document]
Comment (neozhaoliang) [reply docs/lang/articles/reference/global_settings.md:94]: I'm not sure the word "silence" is a good choice here.

File: docs/lang/articles/reference/global_settings.md [hunk 5]
```diff
@@ -4,72 +4,194 @@ sidebar_position: 3
 
 # Global Settings
 
+
+The call `ti.init()` is indispensable in every Taichi program. It allows you to customize your Taichi runtime by passing arguments to it or setting environment variables. Each argument or environment variable controls one specific behavior of the Taichi runtime. For example, the argument `arch` specifies the backend, and the argument `debug` decides whether to run the program in debug mode. The document [Hello, World!](../get-started/index.md) gives a brief introduction to this function, and this document provides more details.
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/reference/global_settings.md:8]:
```suggestion
The call `ti.init()` is indispensable in every Taichi program. Taichi allows you to customize your Taichi runtime by passing arguments to `ti.init()` or setting environment variables. Each argument or environment variable controls one specific behavior of the Taichi runtime. For example, the argument `arch` specifies the backend, and the argument `debug` decides whether to run the program in debug mode. The document [Hello, World!](../get-started/index.md) gives a brief introduction to this function, and this document provides more details.
```

File: docs/lang/articles/reference/global_settings.md [hunk 6]
```diff
@@ -4,72 +4,194 @@ sidebar_position: 3
...
+
+Generally, when calling `ti.init()`, Taichi executes the following steps to initialize a specific configuration. We use the `arch` argument as an example:
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/reference/global_settings.md:10]:
```suggestion
Generally, when you call `ti.init()`, Taichi executes the following steps to initialize a specific configuration. We use the `arch` argument as an example:
```
'calling' takes 'Taichi' as its subject, which is incorrect.

File: docs/lang/articles/reference/global_settings.md [hunk 7]
```diff
@@ -4,72 +4,194 @@ sidebar_position: 3
...
+
+1. Firstly, Taichi looks for the argument that is passed to `ti.init()`. In this case, after Taichi reads `ti.init(arch=cuda)`, it chooses CUDA as the backend and omits the environment variable.
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/reference/global_settings.md:12]:
```suggestion
1. Taichi looks for the arguments passed to `ti.init()`. In this case, after Taichi reads `ti.init(arch=cuda)`, it chooses CUDA as the backend and omits the environment variable.
```
I believe argument should be in plural form here.

File: docs/lang/articles/reference/global_settings.md [hunk 8]
```diff
@@ -4,72 +4,194 @@ sidebar_position: 3
...
+2. If no argument is found, Taichi checks the corresponding environment variable. In this case, if `arch` is not specified but replaced by the environment variable `export TI_ARCH=cuda`, Taichi still chooses CUDA as the backend.
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/reference/global_settings.md:13]:
```suggestion
2. If no argument is found, Taichi checks the corresponding environment variable. In this case, if `arch` is not specified but the environment variable is set to `export TI_ARCH=cuda`, Taichi chooses CUDA as the backend.
```

File: docs/lang/articles/reference/global_settings.md [hunk 9]
```diff
@@ -4,72 +4,194 @@ sidebar_position: 3
...
+3. If no customized setting is found, Taichi uses a default configuration. In this case, if neither the argument `arch` is specified nor the environment variable `TI_ARCH` is found, Taichi adopts the default backend `arch=ti.cpu`.
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/reference/global_settings.md:14]:
```suggestion
3. If no customized setting is found, Taichi uses a default configuration. In this case, if neither the argument `arch` is specified nor an environment variable `TI_ARCH` is found, Taichi adopts the default backend `arch=ti.cpu`.
```

File: docs/lang/articles/reference/global_settings.md [hunk 10]
```diff
@@ -4,72 +4,194 @@ sidebar_position: 3
...
+
+
+Below are some most frequently used configurations you can set with the `ti.init()` call:
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/reference/global_settings.md:17]:
```suggestion
Following are some frequently used configurations that the `ti.init()` method call supports:
```

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]
LGTM! Thank you! Some minor editorial changes.

File: docs/lang/articles/reference/global_settings.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Olinaaaloompa) [reply docs/lang/articles/reference/global_settings.md:100]: Have rephrased. Please check the sentence again.

[the remark this answers is not in the document]
Comment (neozhaoliang) [reply docs/lang/articles/reference/global_settings.md:100]: "Interchangeable" is a bit obscure here. Let's say "not all of them have corresponding arguments of the `init` function"?

**Review by Vissidarte-Herman — APPROVED** [review]
LGTM

**Review by Olinaaaloompa — APPROVED** [review]
LGTM

**Pull request merged**